syntax = "proto3";

package protodrive;

service Drive{
    rpc GetPosition(Direction) returns (stream Point){}
    rpc DriverStop(Direction) returns (StopInfo){}
    rpc ChangeImage(Direction) returns (ChangeInfo){}
    rpc SendSEPosition(BriefPoints) returns(SEInfo){}
}

message Direction{
    string op = 1;
    int32 t = 2;
}
message Point{
    string id = 1;
    float lon = 2;
    float lat = 3;
    float x = 4;
    float y = 5;
    float angle = 6;
    int32 t=7;
}
message BriefPoints{
    float x1 = 1;
    float y1 = 2;
    float x2 = 3;
    float y2 = 4;
}
message StopInfo{
    bool isStop = 1;
    int32 t=2;
}
message ChangeInfo{
    string image = 1;
    int32 t=2;
}
message SEInfo{
    string state = 1;
}
